chase/frontend/components/speakers_list/add_speaker.tsx

```diff
@@ -0,0 +1,168 @@
+ * The user can select a country from the list of suggestions.
+ * The user can click on the Add Speaker Button to add the speaker and keep the overlay open, or click on the Add and Close Button to add the speaker and close the overlay.
+ * Instead of clicking on the Add Speaker Button, the user can press the Enter key after selecting a country to trigger the same action.
+ * Note: Not only countries can be added to the Speakers List, but also Non-State Actors and als UN Staff like the Secretary-General (Country-Code: unm / unw (male/female))
+ */
+
+// TODO add toast functionality for when a speaker is added or already on the list
+// TODO add warning when a speaker is added to the list and the list is closed
+
+export default function AddSpeakerOverlay({
+  listOfAllCountries,
+  listClosed,
+  closeOverlay,
+}: {
+  listOfAllCountries: CountryCode[];
+  listClosed: boolean;
+  closeOverlay: () => void;
+}) {
+  const { LL, locale } = useI18nContext();
+
+  const [countries, setCountries] = useState<CountryData[] | null>(null);
+  const [query, setQuery] = useState<string>("");
+  const [filteredCountries, setFilteredCountries] = useState<CountryData[]>([]);
+  const [selectedCountry, setSelectedCountry] = useState<CountryData | null>(
+    null,
+  );
+
+  useEffect(() => {
+    if (!listOfAllCountries) return;
+    const countryData: CountryData[] = listOfAllCountries.map(
+      (country: CountryCode) => {
+        return {
+          alpha3: country,
+          name: getCountryNameByCode(country, locale),
+        };
+      },
+    );
+    setCountries(countryData);
+  }, [listOfAllCountries, locale]);
+
+  useEffect(() => {
+    if (selectedCountry) {
+      setQuery(selectedCountry.name);
+    }
+  }, [selectedCountry]);
+
+  useEffect(() => {
+    if (query === "") {
+      setSelectedCountry(null);
+    } else {
+      const country = countries?.find((country) => country.name === query);
+      if (country) {
+        setSelectedCountry(country);
+      }
+    }
+  }, [query]);
+
+  const searchCountry = (event: AutoCompleteCompleteEvent) => {
+    if (!countries) return;
+
+    setTimeout(() => {
+      let filteredCountries;
+      if (!event.query.trim().length) {
+        filteredCountries = [...countries];
+      } else {
+        filteredCountries = countries.filter((country) => {
+          return (
+            country.name.toLowerCase().startsWith(event.query.toLowerCase()) ||
+            country.alpha3.toLowerCase().startsWith(event.query.toLowerCase())
+          );
+        });
+      }
+      setFilteredCountries(filteredCountries);
+    }, 250);
+  };
+
+  const countryTemplate = (item: CountryData) => {
+    return (
+      <div className="flex items-center">
+        <SmallFlag countryCode={item.alpha3} />
+        <div className="ml-2">{item.name}</div>
+      </div>
+    );
+  };
+
+  const sendAddSpeaker = () => {
+    if (selectedCountry) {
+      console.log(selectedCountry?.alpha3);
```

Comment:
We probably should add a linter rule do disallow console calls

Reply:
Oh yeah, good idea. But I would rather keep that this way until i dont have to mimic api calls